libs/core/src/main/java/org/elasticsearch/core/RefCounted.java

```diff
@@ -74,4 +74,29 @@ default void mustIncRef() {
         assert false : AbstractRefCounted.ALREADY_CLOSED_MESSAGE;
         incRef(); // throws an ISE
     }
+
+    /**
+     * A noop implementation that always behaves as if it is referenced and cannot be released.
+     */
+    RefCounted ALWAYS_REFERENCED = new RefCounted() {
+        @Override
+        public void incRef() {
+
+        }
```

Comment:
nit: excessive whitespace here

```suggestion
        public void incRef() {}
```